### ISSUE TITLE
About training

### ISSUE THREAD

**Issue by xzjzsa** [mli0603/stereo-transformer#43]
Excuse me, have you tested the results of training directly on the KITTI?

**Comment by mli0603** [comment]
Hi @xzjzsa 

Can you elaborate? Training on KITTI and testing on KITTI or training on Scene Flow and testing on KITTI?

**Comment by xzjzsa** [comment]
@mli0603 
My validation metrics on the KITTI dataset have not declined, is it because I am not trained enough? I've observed that training losses have been going down.

Index 0, l1_raw 34.5487, rr 4.7418, l1 34.7884, occ_be 1.4406, epe 35.2884, iou 0.0214, px error 1.0000
Index 1, l1_raw 31.8053, rr 4.7866, l1 32.0896, occ_be 1.4368, epe 32.5896, iou 0.0133, px error 1.0000
Index 2, l1_raw 34.9226, rr 4.7612, l1 35.1358, occ_be 1.4041, epe 35.6358, iou 0.0243, px error 1.0000
Index 3, l1_raw 37.5268, rr 4.8069, l1 37.8074, occ_be 1.3815, epe 38.3074, iou 0.0223, px error 1.0000
Index 4, l1_raw 52.1304, rr 4.8323, l1 52.1313, occ_be 1.3878, epe 52.6313, iou 0.0174, px error 1.0000
Index 5, l1_raw 47.6444, rr 4.7991, l1 47.9339, occ_be 1.5542, epe 48.4339, iou 0.0153, px error 0.9999
Index 6, l1_raw 36.9510, rr 4.7818, l1 37.3028, occ_be 1.4805, epe 37.8028, iou 0.0182, px error 1.0000
Index 7, l1_raw 37.5950, rr 4.7931, l1 37.7550, occ_be 1.3953, epe 38.2549, iou 0.0152, px error 0.9993
Index 8, l1_raw 31.6898, rr 4.7799, l1 32.2496, occ_be 1.4290, epe 32.7496, iou 0.0249, px error 0.9996
Index 9, l1_raw 44.5085, rr 4.7719, l1 45.0399, occ_be 1.4564, epe 45.5399, iou 0.0220, px error 1.0000
Epoch 113, epe 39.7234, iou 0.0194, px error 0.9999

**Comment by xzjzsa** [comment]
@mli0603 There's no change in my eval metrics.Is it because I don't train enough?
Epoch 0, epe 108.2090, iou 0.9518, px error 0.9772
...
Epoch 12, epe 21.3825, iou 0.9694, px error 0.8998
...
Epoch 112, epe 21.0324, iou 0.9765, px error 0.8987

**Comment by xzjzsa** [comment]
@mli0603  I‘m training on KITTI.

**Comment by xzjzsa** [comment]
> Hi @xzjzsa
> 
> Sorry I didn't get your question. Can you elaborate? Training on KITTI and testing on KITTI or training on Scene Flow and testing on KITTI?
> 
> * For training on KITTI and submit to KITTI leaderboard, our result is summarized in Table 4
> * For training on Scene Flow and test on KITTI training data, our result is summarized in Table 2

What if instead of pre-training at SceneFlow, you train directly at KITTI?

**Comment by xzjzsa** [comment]
Hi @mli0603 
In my training log:
epoch 14
Index 199, l1_raw 10.9556, rr 0.9234, l1 7.4534, occ_be 0.0198, epe 7.7481, iou 0.9848, px error 0.2480
But when I use the saved model to inference,the result is:
3 px error 28.398%
EPE 16.817174
Do you know the reason?

**Comment by mli0603** [comment]
Hi @xzjzsa,

My understanding is that you are training on SceneFlow and you are getting 7.75 EPE on SceneFlow? And you are saying that provided SceneFlow model gives you 16.82 EPE on SceneFlow?

This is odd and I would say something is wrong with the data. Have you checked the output disparity and seen anything odd?

**Comment by xzjzsa** [comment]
Hi @mli0603 
For the same image, the test results during training are different from the inference results of the model saved after the training.I don't know why.

**Comment by mli0603** [comment]
Hi @xzjzsa 

Oh I see what you mean now. Have you seen this issue #8? In short, torch version will lead to your problem. I assume you are not using torch 1.5.1?

**Comment by xzjzsa** [comment]
Hi @mli0603 
my torch version is torch 1.5.1. Is there any error in my inference code for reading SceneFlow data?

left = np.array(Image.open('./sample_data/SCENE_FLOW/RGB_cleanpass/left/A-0032-0006.png').convert('RGB'))
right = np.array(Image.open('./sample_data/SCENE_FLOW/RGB_cleanpass/right/A-0032-0006.png').convert('RGB'))
disp,_ = readPFM("./sample_data/SCENE_FLOW/disparity/left/A-0032-0006.pfm")
input_data = {'left': left, 'right':right, 'disp':disp}
input_data = normalization(**input_data)

**Comment by mli0603** [comment]
Hmm, can you share the data with me? Maybe dump the images and disparity and share me a link so I can take a look
